examples/datacalc.py

```diff
@@ -94,38 +99,42 @@ def updateDB(self, busName: str, msg: str):
             except json.JSONDecodeError as e:
                 print(f"apps.datacalc.updateDB(): {e!r}")
             else:
-                orgDbNames = self.dbNames.copy()
-                self.dbs = {"": ""}
-                for dbBox in self.viewerFrame.dbBoxes.values():
-                    dbBox.clear()
-                    dbBox.addItem("")
+                originalDBs = set(self.dbs)
+                newDBs = set([""])
                 for db in msg.get("db", ()):
                     if all(key in db for key in ("name", "path")):
                         name, path = db["name"], db["path"]
-                        self.dbs[name] = path
-                        for dbBox in self.viewerFrame.dbBoxes.values():
-                            dbBox.addItem(name)
+                        newDBs.add(name)
+                        if name not in self.dbs:
+                            self.dbs[name] = path
+                            for dbBox in self.viewerFrame.dbBoxes.values():
+                                dbBox.addItem(name)
                     else:
                         print(f"The message was ignored because "
                               f"the database {db} has no such key; name or path.")
-                for name, orgDbName in orgDbNames.items():
-                    if orgDbName in self.dbs:
-                        self.viewerFrame.dbBoxes[name].setCurrentText(orgDbName)
+                removingDBs = originalDBs - newDBs
+                for dbBox in self.viewerFrame.dbBoxes.values():
+                    if dbBox.currentText() in removingDBs:
+                        dbBox.setCurrentText("")
```

Comment:
Why do you need to make the text ""?

Reply:
In `__init__()`, the initial value of `dbBox` is `""`, an empty string, which represents that no db is selected.

So, if the selected db is removed, it should be updated `""`. Otherswise, it will be reselected as an unintended db.